tests/runtests.bats

```diff
@@ -215,3 +215,64 @@ cleanup() {
 
   cleanup
 }
+
+@test "bring up and save a snapshot" {
+  export VAGRANT_CWD=tests/snapshot
+
+  cleanup
+  run ${VAGRANT_CMD} up ${VAGRANT_OPT}
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot save default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot list default 
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot delete default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+
+  cleanup
+}
+
+@test "bring up and save a snapshot and restore it" {
+  export VAGRANT_CWD=tests/snapshot
+
+  cleanup
+  run ${VAGRANT_CMD} up ${VAGRANT_OPT}
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot save default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot list default 
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} ssh -- -t 'touch a.txt'
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot restore default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} ssh -- -t 'ls a.txt'
+  echo "${output}"
+  echo "status = ${status}"
+  # This means that the file does not exist on the box.
+  [ "$status" -eq 1 ]
+  run ${VAGRANT_CMD} snapshot delete default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+
+  cleanup
+}
```

Comment:
@electrofelix I tried to implement the integration tests you mentioned in https://github.com/vagrant-libvirt/vagrant-libvirt/pull/1456#pullrequestreview-950098062, let me know if something is off in my understanding of it.